ignite-core/src/main/java/io/micronaut/ignite/annotation/IgniteTransaction.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.ignite.annotation;
+
+import io.micronaut.aop.Around;
+import io.micronaut.context.annotation.Type;
+import io.micronaut.ignite.intercept.IgniteTransactionInterceptor;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * wraps a call in an ignite transaction.
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
+@Around
+@Type(IgniteTransactionInterceptor.class)
+public @interface IgniteTransaction {
```

Comment:
Another thought, do we really need this interface? Application developers can get access to the streaming transactional capabilities of Ignite by injecting an instance of `Ignite` (created by your integration) into their code and by using the `ignite.transactions(..)` method next.